### PR TITLE
Cleaned up Contact Information Flipper in specs

### PR DESCRIPTION
**Pull request by RachalCassity**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

Cleaned up spec files due to a comment made in this [PR](https://github.com/department-of-veterans-affairs/vets-api/pull/18729)

## Related issue(s)

- department-of-veterans-affairs/va.gov-team/issues/73776

## Testing done

- [X] *New code is covered by unit tests*
- No new behavior added. Cleaned up spec tests
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: lib/va_profile/military_personnel/service.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def get_service_history
 
           response = perform(:post, identity_path, VAProfile::Models::ServiceHistory.in_json)
 
-          ServiceHistoryResponse.from(@current_user, response)
+          ServiceHistoryResponse.from(@user, response)
```
Comment (RachalCassity) [inline lib/va_profile/military_personnel/service.rb:34]: Found this bug. @current_user is not defined but @user is.

File: modules/veteran/spec/sidekiq/representatives/update_spec.rb [hunk 2]
```diff
@@ -731,7 +717,7 @@ def create_flagged_records(flag_type)
       end
 
       before do
-        Flipper.enable(:va_v3_contact_information_service)
+        allow(Flipper).to receive(:enabled?).with(:va_v3_contact_information_service).and_return(true)
```
Comment (rmtolmach) [inline modules/veteran/spec/sidekiq/representatives/update_spec.rb:720]: Why did you update this line to enable the flipper for this block but delete `Flipper.enable(:va_v3_contact_information_service)` in other places in this file?

Comment (RachalCassity) [reply modules/veteran/spec/sidekiq/representatives/update_spec.rb:720]: Enabling a flipper can cause other spec tests to fail. After you enable the Flipper, you have to disable this.  `allow(Flipper).to receive(:enabled?).with(:va_v3_contact_information_service).and_return(true)` will only enable the flipper for specific tests.

File: spec/requests/v0/profile/addresses_spec.rb [hunk 3]
```diff
@@ -348,322 +332,296 @@
...
+            end.to change(AsyncTransaction::VAProfile::AddressTransaction, :count).from(0).to(1)
+          end
         end
       end
     end
-  end
-
-  describe 'POST /v0/profile/addresses 400 v2' do
-    before do
-      Flipper.enable(:va_v3_contact_information_service)
```
Comment (rmtolmach) [inline spec/requests/v0/profile/addresses_spec.rb:393]: You don't need `allow(Flipper).to receive(:enabled?)...` for this section?

File: spec/requests/v0/profile/addresses_spec.rb [hunk 4]
```diff
@@ -348,322 +332,296 @@
...
+          expect(response).to match_camelized_response_schema('errors')
+          expect(errors_for(response)).to include "address-pou - can't be blank"
+        end
       end
     end
-  end
-
-  describe 'PUT /v0/profile/addresses v2' do
-    before do
-      Flipper.enable(:va_v3_contact_information_service)
```
Comment (rmtolmach) [inline spec/requests/v0/profile/addresses_spec.rb:500]: Same question here: you don't need `allow(Flipper).to receive(:enabled?)...` for these tests?

File: spec/requests/v0/profile/addresses_spec.rb [hunk 5]
```diff
@@ -348,322 +332,296 @@
...
-        expect(errors_for(response)).to include 'address - must contain ASCII characters only'
+          put('/v0/profile/addresses', params: address.to_json, headers:)
+        end
       end
-    end
-  end
 
-  describe 'DELETE /v0/profile/addresses v2' do
-    before do
-      Flipper.enable(:va_v3_contact_information_service)
```
Comment (rmtolmach) [inline spec/requests/v0/profile/addresses_spec.rb:609]: And here as well.

File: spec/requests/v0/profile/addresses_spec.rb (hunk 4 above)
Comment (RachalCassity) [reply spec/requests/v0/profile/addresses_spec.rb:500]: The flipper is enabled on [line 337](https://github.com/department-of-veterans-affairs/vets-api/blob/b4ed91d588872d777815af3919818d2668e4515b/spec/requests/v0/profile/addresses_spec.rb#L337)

File: spec/requests/v0/profile/addresses_spec.rb (hunk 5 above)
Comment (RachalCassity) [reply spec/requests/v0/profile/addresses_spec.rb:609]: The flipper is enabled on [line 337](https://github.com/department-of-veterans-affairs/vets-api/blob/b4ed91d588872d777815af3919818d2668e4515b/spec/requests/v0/profile/addresses_spec.rb#L337)

File: modules/veteran/spec/sidekiq/representatives/update_spec.rb (hunk 2 above)
Comment (rmtolmach) [reply modules/veteran/spec/sidekiq/representatives/update_spec.rb:720]: I'm wondering why you didn't need to call that `allow(Flipper)...` code further down in this file, like here: 
image

File: spec/requests/v0/profile/addresses_spec.rb (hunk 4 above)
Comment (rmtolmach) [reply spec/requests/v0/profile/addresses_spec.rb:500]: ohh, okay. I see now. It was hard to discern the indentation.

File: spec/requests/v0/profile/addresses_spec.rb (hunk 3 above)
Comment (RachalCassity) [reply spec/requests/v0/profile/addresses_spec.rb:393]: @rmtolmach It's on line 378

File: modules/veteran/spec/sidekiq/representatives/update_spec.rb (hunk 2 above)
Comment (RachalCassity) [reply modules/veteran/spec/sidekiq/representatives/update_spec.rb:720]: Line 720 will take automatically enable the flipper for 841 and 1029 :)

Comment (RachalCassity) [reply modules/veteran/spec/sidekiq/representatives/update_spec.rb:720]: @rmtolmach I removed the need for the repetitive `Flipper.enable` calls

Comment (RachalCassity) [reply modules/veteran/spec/sidekiq/representatives/update_spec.rb:720]: All tests that had the flipper enabled are still enabled. I just combined encased all the tests

**Review by rmtolmach — APPROVED** [review]
👍

I've updated the [features toggle guide example](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#FeatureTogglesGuide-Backendexample) to show this mocking method.

**Pull request merged**